### PR TITLE
Server audit dialog (master)

### PR DESCRIPTION
**Pull request by eszkadev**

### CODE REVIEW

File: browser/src/control/jsdialog/Widget.TreeView.js [hunk 1]
```diff
@@ -30,7 +30,7 @@
  *     headers: [ { text: 'first column' }, { text: 'second' }],
  *     entries: [
  *         { row: 0, columns [ { text: 'a' }, { collapsed: 'collapsedIcon.svg' }, { collapsedimage: '<BASE64 encoded PNG>' } ] },
- *         { row: 1, columns [ { text: 'a2' }, { expanded: 'expandedIcon.svg' }, selected: true ]}
+ *         { row: 1, columns [ { link: 'http://someurl' }, { expanded: 'expandedIcon.svg' }, selected: true ]}
```
Comment (gokaysatir) [inline browser/src/control/jsdialog/Widget.TreeView.js:33]: Below address is used for this purpose, just fyi.
https://example.com/

Comment (eszkadev) [reply browser/src/control/jsdialog/Widget.TreeView.js]: done

File: wsd/ServerAuditUtil.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    for (auto entry : entries)
+        result += "{\"code\": \"" + entry.first + "\", \"status\": \"" + entry.second + "\"}";
+
+    result += "]}";
+    return result;
+}
+
+void ServerAuditUtil::addEntry(std::string code, std::string status)
+{
+    entries.push_back(std::make_pair(code, status));
```
Comment (gokaysatir) [inline wsd/ServerAuditUtil.cpp:29]: Is this thread safe?

**Review by caolanm — APPROVED** [review]

File: wsd/ServerAuditUtil.cpp (hunk 2 above)
Comment (eszkadev) [reply wsd/ServerAuditUtil.cpp]: moved static map to DocumentBroker now, so it is not shared

**Comment by caolanm** [comment]
great, much more reassuring :-)

**Pull request merged**